InventoryConfigurableProduct/Plugin/Model/ResourceModel/Product/Indexer/Price/Configurable/FilterSelectByInventoryPlugin.php

```diff
@@ -0,0 +1,106 @@
+
+use Magento\CatalogInventory\Model\Stock;
+use Magento\ConfigurableProduct\Model\ResourceModel\Product\Indexer\Price\Configurable;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\DB\Select;
+
+/**
+ * Plugin for FilterSelectByInventory to add "is_salable" filter.
+ */
+class FilterSelectByInventoryPlugin
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @var StockIndexTableNameResolverInterface
+     */
+    private $stockIndexTableNameResolver;
+
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param StockResolverInterface $stockResolver
+     * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        StockResolverInterface $stockResolver,
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider = null
+    ) {
+        $this->storeManager = $storeManager;
+        $this->stockResolver = $stockResolver;
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->defaultStockProvider = $defaultStockProvider ?: ObjectManager::getInstance()
+            ->get(DefaultStockProviderInterface::class);
+    }
+
+    /**
+     * Add "is_salable" filter to select.
+     *
+     * @param Configurable $subject
+     * @param Select $select
+     * @return Select
+     *
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterFilterSelectByInventory(Configurable $subject, Select $select): Select
+    {
+        $websiteCode = $this->storeManager->getWebsite()->getCode();
+        $stock = $this->stockResolver->execute(SalesChannelInterface::TYPE_WEBSITE, $websiteCode);
+        $stockId = (int)$stock->getStockId();
+        if ($stockId === $this->defaultStockProvider->getId()) {
+            return $select;
+        }
+        $stockTable = $this->stockIndexTableNameResolver->execute($stockId);
+
+        $whereParts = $select->getPart(Select::WHERE);
+        foreach($whereParts as $key => $cond){
```

Comment:
It's weird to iterate where condition and remove statements from it. Is it possible to add new logic in another way?